### PR TITLE
Add IPAddress ISpanFormattable/Parsable implementations

### LINKED ISSUE #82842: [API Proposal]: IPAddress should implement ISpanFormattable/Parsable<IPAddress>
### Background and motivation

IPAddress already provides TryFormat/TryParse methods.  It should implement the appropriate interfaces to go along with these.  Doing that for `IPAddress`, for example, means the `IPAddress`'s `TryFormat` will be used as part of string interpolation rather than its `ToString()` being used.

### API Proposal

```diff
namespace System.Net

public class IPAddress
+    : ISpanFormattable, ISpanParsable<IPAddress>
{
    // all of the interface members implemented explicitly
}
```

### API Usage

```C#
IPAddress address = ...;
string s = $"The address is {address}";
```

### Alternative Designs

_No response_

### Risks

_No response_

### PR DESCRIPTION
**Pull request by stephentoub**
Fixes https://github.com/dotnet/runtime/issues/82842

### CODE REVIEW

File: src/libraries/System.Net.Primitives/src/System/Net/IPAddress.cs [hunk 1]
```diff
@@ -236,6 +236,14 @@ public static bool TryParse(ReadOnlySpan<char> ipSpan, [NotNullWhen(true)] out I
             return (address != null);
         }
 
+        static bool IParsable<IPAddress>.TryParse([NotNullWhen(true)] string? s, IFormatProvider? provider, [NotNullWhen(true)] out IPAddress? result) =>
```
Comment (antonfirsov) [inline src/libraries/System.Net.Primitives/src/System/Net/IPAddress.cs:239]: Shouldn't we add  `/// <inheritdoc />` to all these?

Comment (stephentoub) [reply src/libraries/System.Net.Primitives/src/System/Net/IPAddress.cs:242]: We can, though it doesn't really help for these.

**Review by MihaZupan — APPROVED** [review]

**Review by antonfirsov — APPROVED** [review]

**Comment by stephentoub** [comment]
@lewing, can you help me understand what would cause this failure on wasm?
```
[22:01:25] info: Discovering: System.Net.Primitives.Pal.Tests.dll (method display = ClassAndMethod, method display options = None)
[22:01:25] info:    at System.RuntimeType.GetMethodsByName(String , BindingFlags , MemberListType , RuntimeType )
[22:01:25] info:    at System.RuntimeType.GetMethodCandidates(String , BindingFlags , CallingConventions , Type[] , Int32 , Boolean )
[22:01:25] info:    at System.RuntimeType.GetMethods(BindingFlags )
[22:01:25] info:    at System.Reflection.RuntimeReflectionExtensions.GetRuntimeMethods(Type )
[22:01:25] info:    at Xunit.Sdk.ReflectionTypeInfo.GetMethods(Boolean )
[22:01:25] info:    at Xunit.Sdk.XunitTestFrameworkDiscoverer.FindTestsForType(ITestClass , Boolean , IMessageBus , ITestFrameworkDiscoveryOptions )
[22:01:25] info:    at Microsoft.DotNet.XHarness.TestRunners.Xunit.ThreadlessXunitDiscoverer.FindWithoutThreads(Boolean , IMessageSink , ITestFrameworkDiscoveryOptions )
[22:01:25] info:    at Microsoft.DotNet.XHarness.TestRunners.Xunit.ThreadlessXunitTestRunner.Run(IEnumerable`1 )
[22:01:25] info:    at Microsoft.DotNet.XHarness.TestRunners.Common.ApplicationEntryPoint.InternalRunAsync(LogWriter )
[22:01:25] info:    at Microsoft.DotNet.XHarness.TestRunners.Common.ApplicationEntryPoint.InternalRunAsync(ApplicationOptions , TextWriter , TextWriter )
[22:01:25] info:    at Microsoft.DotNet.XHarness.TestRunners.Common.WasmApplicationEntryPointBase.RunAsync()
[22:01:25] info:    at Microsoft.DotNet.XHarness.TestRunners.Xunit.WasmApplicationEntryPoint.Run()
[22:01:25] info:    at SimpleWasmTestRunner.Main(String[] args)
[22:01:25] info: Error: Could not load list of method overrides due to Method not found: bool System.IParsable`1.TryParse(string,System.IFormatProvider,!0&)
[22:01:25] info:     at To (/root/helix/work/workitem/e/wasm_build/AppBundle/dotnet.js:3:104514)
[22:01:25] info:     at jo (/root/helix/work/workitem/e/wasm_build/AppBundle/dotnet.js:3:103879)
[22:01:25] info:     at do_icall (do_icall (wasm://wasm/00a9a02e:wasm-function[376]:0x208a7))
[22:01:25] info:     at do_icall_wrapper (do_icall_wrapper (wasm://wasm/00a9a02e:wasm-function[246]:0x1d2db))
[22:01:25] info:     at mono_interp_exec_method (mono_interp_exec_method (wasm://wasm/00a9a02e:wasm-function[236]:0x10fa5))
[22:01:25] info:     at interp_runtime_invoke (interp_runtime_invoke (wasm://wasm/00a9a02e:wasm-function[284]:0x1e2ba))
[22:01:25] info:     at mono_jit_runtime_invoke (mono_jit_runtime_invoke (wasm://wasm/00a9a02e:wasm-function[8470]:0x1b872f))
[22:01:25] info:     at do_runtime_invoke (do_runtime_invoke (wasm://wasm/00a9a02e:wasm-function[2815]:0xae576))
[22:01:25] info:     at mono_runtime_try_invoke (mono_runtime_try_invoke (wasm://wasm/00a9a02e:wasm-function[2820]:0xaec18))
[22:01:25] info:     at mono_runtime_invoke (mono_runtime_invoke (wasm://wasm/00a9a02e:wasm-function[2847]:0xb0f0f))
[22:01:25] info: WASM EXIT 1
[22:01:25] info: exiting due to exception: Error: Could not load list of method overrides due to Method not found: bool System.IParsable`1.TryParse(string,System.IFormatProvider,!0&)
[22:01:25] info: Process v8 exited with 1
```
The cited method is an explicit static interface implementation added in this PR.

**Comment by lewing** [comment]
I'll take a look but failures in the EAT (Enable Aggressive Trimming) lane are almost always trimming related

**Comment by lewing** [comment]
@lambdageek @vitek-karas @sbomer any idea why class initialization would fail here after the aggressive trimming pass?

**Comment by jtschuster** [comment]
The error looks the same as the failures that https://github.com/dotnet/runtime/pull/82197 fixed, but since all the assemblies are fully trimmed, I think this might be related to having multiple layers with IPAddress implementing ISpanParsable which implements IParsable. I'll look into it.

**Comment by stephentoub** [comment]
Thanks. In the meantime, can I disable these test suites for these legs, and if so, what's the best way to do so currently?  Or should I hold off on this PR until you complete your investigation?

**Comment by stephentoub** [comment]
@lewing, I merged in the latest main and it's now passed the wasm legs.  Does that mean this is ok to merge?  Did we maybe take in some dependency update that included a fix?

**Comment by lewing** [comment]
go ahead and merge, i'll keep an eye out for any problems

**Comment by jtschuster** [comment]
> Did we maybe take in some dependency update that included a fix?

Looks like yes, the failure was using version 8.0.0-preview.2.23112.4 of the linker which didn't have https://github.com/dotnet/runtime/pull/82197, and now runtime uses version 8.0.0-preview.2.23126.3 with the fix.

**Pull request merged**